src/CSnakes.Runtime/CPython/Init.cs

```diff
@@ -138,6 +138,15 @@ protected virtual void Dispose(bool disposing)
                         return;
 
                     Debug.WriteLine("Calling Py_Finalize()");
+
+                    // Acquire the GIL only to dispose it immediately because `PyGILState_Release`
+                    // is not available after `Py_Finalize` is called. This is done primarily to
+                    // trigger the disposal of handles that have been queued before the Python
+                    // runtime is finalized.
+
+                    GIL.Acquire().Dispose();

```

Comment:
In future, it would be good to have a GIL.Finalize() method which does the same thing. Whilst this is only done in 1 place, it's non-obvious.